components/chat/useChatSocket.tsx

```diff
@@ -264,8 +267,27 @@ const useChatSocket = (isEmpty?: boolean) => {
     []
   );
 
-  const handleToolAdjusted = (tools: string[]) => {
+  const handleToolChange = (
```

Comment:
I will probably make this an async function. .then syntax is too hard to use and read.

Reply:
Done.